File: RandomData.r


setwd("/path/to/working/directory")

epoch_time <- as.integer(Sys.time())
as.POSIXct(epoch_time, origin="1970-01-01")

lapply(seq(10), function(i) {
  df <- data.frame(
       ID = replicate(50, sample(seq(15), 1, replace=TRUE)),
       GROUP = replicate(50, sample(LETTERS, 1, replace=TRUE)),
       DATE = as.POSIXct(replicate(50, sample(epoch_time, 1, replace=TRUE)), origin="1970-01-01"),
       NUM1 = rnorm(50)*100,
       NUM2 = rnorm(50),
       BOOL = replicate(50, sample(c("TRUE", "FALSE"), 1, replace=TRUE))
  )
 write.csv(df, paste0("RandomData_R_",i,".csv"))
 return(df)
})


str(df)

File: RandomData.py

import os
import numpy as np
import pandas as pd
import datetime as dt
import time

cd = os.path.dirname(os.path.abspath(__file__))
LETTERS = list('ABCDEFGHIJKLMNOPQRSTUVWXYZ')

epoch_time = int(time.time())


for i in range(1,11):
    df = pd.DataFrame({'ID': [np.random.randint(15) for _ in range(50)],
                       'GROUP': ["".join(np.random.choice(LETTERS,1)) for _ in range(50)],
                       'NUM1': np.random.randn(50)*100,
                       'NUM2': np.random.randn(50),
                       'BOOL': [np.random.choice([True,False],1).item(0) for _ in range(50)],
                       'DATE': [dt.datetime.fromtimestamp(np.random.randint(epoch_time)) for _ in range(50)]},
                      columns = ['ID','GROUP','DATE','NUM1','NUM2','BOOL'])
                       
    df.to_csv(os.path.join(cd, 'RandomData_PY_'+str(i)+'.csv'))


